hcl/writer.go

```diff
@@ -214,15 +214,14 @@ func walk(dest, src reflect.Value, interpolate map[string]string, name string, r
 }
 
 // isMutualInterpolation will simply go through the list of relations to find out
-// if a relation is already present between the two resources
-func isMutualInterpolation(target, source string, relations *[][]string) bool {
-	for _, r := range *relations {
-		if (r[0] == target &&
-			r[1] == source) ||
-			(r[1] == target &&
-				r[0] == source) {
-			return true
-		}
+// if a relation is already present between the two resources in one direction
+// or the other
+func isMutualInterpolation(target, source string, relations *map[string]struct{}) bool {
+	if _, ok := (*relations)[fmt.Sprintf("%s+%s", source, target)]; ok {
+		return true
+	}
+	if _, ok := (*relations)[fmt.Sprintf("%s+%s", target, source)]; ok {
+		return true
```

Comment:
We actually do not need the `+` on the interpolation, it's more up to you if you want.

Reply:
what do you mean ? the `+` is just a way to create the key in the `relations` map :thinking: